### PR TITLE
Added support for detection of tercetos encadenados

### PR DESCRIPTION
**Pull request by alvp**

### CODE REVIEW

File: src/rantanplan/rhymes.py [hunk 1]
```diff
@@ -228,8 +228,13 @@ def search_structure(rhyme, rhythmical_lengths, structure_key, structures=None):
     if structures is None:
         structures = STRUCTURES
     for index, (key, _, structure, func) in enumerate(structures):
+        if callable(structure):
+            structure_check = structure(rhyme)
+        else:  # it's a regex
+            structure_re = re.compile(structure, re.VERBOSE)
+            structure_check = structure_re.fullmatch(rhyme)
```
Comment (versae) [inline src/rantanplan/rhymes.py:236]: The docstring needs to be updated.
```
consonance, a rhyme pattern (regex or callable), and a condition on the lengths of sylalbles
```

File: src/rantanplan/structures.py [hunk 2]
```diff
@@ -231,6 +231,32 @@ def is_ovillejo(lengths):
...
+    :param string: String with the rhyme pattern
+    :return: A string with the count values
+    """
+    count_dict = {}
+    output = ""
+    for character in string:
+        count = count_dict.get(character, 0)
+        output += str(count)
+        count_dict[character] = count + 1
+    return output
```
Comment (versae) [inline src/rantanplan/structures.py]: The one liner ICYMI:
```
from collections import defaultdict
counter = defaultdict(int)                                                                                                                   
return "".join([str(counter.update({c: counter[c] + 1}) or counter[c] - 1) for c in s])
```
I would make this function return the plain list of numbers.

Not sure the function name is descriptive enough.

**Review by versae — CHANGES_REQUESTED** [review]

File: src/rantanplan/rhymes.py (hunk 1 above)
Comment (alvp) [reply src/rantanplan/rhymes.py:236]: True, done

File: src/rantanplan/structures.py [hunk 3]
```diff
@@ -231,6 +231,32 @@ def is_ovillejo(lengths):
...
+    Count how many times does a character occur in a given string before its
+    own position.
+    :param string: String with the rhyme pattern
+    :return: A string with the count values
+    """
+    count_dict = {}
+    output = []
+    for character in string:
+        count = count_dict.get(character, 0)
+        output.append(str(count))
```
Comment (versae) [inline src/rantanplan/structures.py]: Any reason why this function needs to return the counts as strings? I know it's used in the main function, but maybe it'd be better to return numbers and have the main function do what it needs with the counts?

File: src/rantanplan/structures.py [hunk 4]
```diff
@@ -231,6 +231,32 @@ def is_ovillejo(lengths):
     return correct_lines == number_of_verses
 
 
+def rank_character_occurrences(string):
```
Comment (versae) [inline src/rantanplan/structures.py]: Aha! What about `get_rhyme_pattern_counts()`?

**Review by versae — APPROVED** [review]

**Pull request merged**